subprojects/dependency-management/src/main/java/org/gradle/internal/component/model/AttributeConfigurationSelector.java

```diff
@@ -158,7 +159,20 @@ private static List<VariantGraphResolveState> findVariantsProvidingExactlySameCl
 
     private static VariantSelectionResult singleVariant(boolean variantAware, List<VariantGraphResolveState> matches) {
         assert matches.size() == 1;
-        return new VariantSelectionResult(ImmutableList.of(matches.get(0)), variantAware);
+
+        VariantGraphResolveState match = matches.get(0);
+        VariantGraphResolveMetadata matchMetadata = matches.get(0).getMetadata();
+
+        if ((matchMetadata instanceof ConfigurationGraphResolveMetadata) &&
+            ((ConfigurationGraphResolveMetadata) matchMetadata).isDeprecatedForConsumption()
+        ) {
+            DeprecationLogger.deprecateConfiguration(matchMetadata.getName())
+                .forConsumption()
+                .willBecomeAnErrorInGradle9()
+                .withUserManual("dependencies_should_no_longer_be_declared_using_the_compile_and_runtime_configurations")
```

Comment:
❌   Is this the right user guide section to link to?  This isn't declaration, and how do we know one of these deprecated configurations is involved?  

[The text of this section](https://docs.gradle.org/current/userguide/upgrading_version_5.html#dependencies_should_no_longer_be_declared_using_the_compile_and_runtime_configurations) seems completely unhelpful in this case.

Reply:
This is the same link we already link to for the existing deprecation. Not saying its right but I'm not sure we have a better place to link to right now.

Reply:
@lkasso Do you have any thoughts on if a better section exists in the docs to link this  error to, or if one should be written?

The situation is the second one Justin noted: Depending on and selecting on a configuration that is deprecated for consumption.

Reply:
The current link is definitely bad. I think this is a sort of unique situation since this deprecation is "ongoing" and isn't necessarily linked to a certain version. There could be many different reasons why a given configuration is deprecated for consumption, so it's difficult to have a link here which genericly works for all configurations.

Reply:
I've been reviewing the documentation, and we don't have an adequate page for this. The current link is old but it's the best we have for now. I will add this to my to-do list and mention it the #bt-docs slack channel when ready.